text/3305-static-in-pattern.md

```diff
@@ -283,6 +282,8 @@ Allowing unsafe-to-access statics in patterns (`static mut`s, untrusted `extern`
 - It's not clear where the `unsafe` keyword would go (within the pattern? around the whole `match` or `let`? what about patterns in function parameters?)
 - it requires Rust to commit to and document, and users to understand, when exactly it is allowed to dereference the static when performing a pattern match
 
+Another alternative is to add a new kind of pattern for runtime equality comparisons, with its own dedicated syntax. In addition to making the language grammar more complex, this option would prevent consts from being interchangeable with statics in pattern matches.
+
```

Comment:
if you wrap constants in such a runtime read syntax, you can hotswap the constant with a static and it will keep working